source/extensions/filters/common/rbac/matchers/upstream_ip_port.cc

```diff
@@ -15,8 +15,10 @@ namespace Matchers {
 using namespace Filters::Common::RBAC;
 
 UpstreamIpPortMatcher::UpstreamIpPortMatcher(
-    const envoy::extensions::rbac::matchers::upstream_ip_port::v3::UpstreamIpPortMatcher& proto)
-    : range_(Network::Address::CidrRange::create(proto.upstream_ip())) {
+    const envoy::extensions::rbac::matchers::upstream_ip_port::v3::UpstreamIpPortMatcher& proto) {
+  if (proto.has_upstream_ip()) {
```

Comment:
Per previous discussion, this needs to fail if neither are set. Please also add a test. Also, please update the docs for the matcher to make it more clear that while both are optional, one or both have to be set.

Reply:
The current logic is such that the matcher will evaluate to fail when neither is set. There is a test already in `rbac_filter_test.cc` - `EmptyUpstreamConfigPolicyDeny` which tests this condition. Please let me know if that is ok.

Reply:
I don't think that's the right behavior. Just fail config load if it's configured this way.

Reply:
You should just be able to throw an exception which will abort the config load.